app-lifelike/src/main/java/com/example/lifelike/composable/loggedin/PhotosOfAlbum.kt

```diff
@@ -84,8 +79,43 @@ interface PhotosOfAlbum {
             val image = imageResource(R.drawable.placeholder)
 
 
+            val photoRows =  mutableListOf<List<Photo>>()
+
+            for (i in album.photos.indices step cols) {
+                val row = mutableListOf<Photo>()
+                for (r in 0 until cols){
+                    if (i+r >= album.photos.size) break
+                    row.add(album.photos[i+r])
+                }
+                photoRows.add(row)
+            }
+
             Box(modifier = Modifier.padding(4.dp)) {
-                Table(columns = cols, columnWidth = { TableColumnWidth.Fraction(1.0f / cols) }) {
+                //scrolling fast may cause exception: https://issuetracker.google.com/issues/154653504
+                AdapterList(
+                        data = photoRows,
+                        itemCallback = { row ->
+                            WithConstraints {
+                                Row {
+                                    val w = with(DensityAmbient.current) { (constraints.maxWidth.toDp().value / cols).dp }
+                                    row.forEach { photo ->
+                                        Box(modifier = Modifier
+                                                .width(w)
+                                                .padding(4.dp)
+                                                .clickable(onClick = { onPhotoSelected(photo) })
+                                        ) {
+                                            Box(modifier = Modifier.aspectRatio(1f).fillMaxSize()) {
+                                                Image(image)
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                )
+                /*Table layout was removed temporarily until we will make it available again with a refreshed API. (Id88a7)
```

Comment:
I suggest to remove old API completely, as it will be refreshed anyway
@zsoltk what do you think?